clients/src/main/java/org/apache/kafka/common/record/FileRecords.java

```diff
@@ -209,8 +223,18 @@ public void setFile(File file) {
      */
     public void renameTo(File f) throws IOException {
         try {
-            channel.close(); 
-            Utils.atomicMoveWithFallback(file.toPath(), f.toPath());
+            if(OperatingSystem.IS_WINDOWS){
+                channel.close();
+
+                Utils.atomicMoveWithFallback(file.toPath(), f.toPath());
+
+                if(!f.getName().endsWith(DeletedFileSuffix)){
```

Comment:
This deleted filtering is a little bizarre. I understand it is just for reopen the .deleted file for deletion. We need to add some sort of comments if this is to be merged to opensource.